packages/babel-plugin-transform-bigint/src/index.js

```diff
@@ -0,0 +1,41 @@
+import syntaxBigInt from "babel-plugin-syntax-bigint";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxBigInt,
+
+    visitor: {
+      Program: {
+        exit(path: NodePath, state: Object) {
+          state.file.addImport("bigint-polyfill", "default", "BigInt");
+        },
+      },
+      BinaryExpression(path) {
+        if (path.node.operator === "+") {
+          if (
+            path.node.left.type === "BigIntLiteral" &&
+            path.node.left.type === "BigIntLiteral"
+          ) {
```

Comment:
Following Dan's example, what I have in mind is something like that:

```
function add(a, b) {
  var tmp;
  if (a instanceof _BingInt || b instanceof _BingInt)
    tmp = _BigIntAdd(a, b)
  else
    tmp = a + b;
  
  return tmp;
}
```

PS.: The names are just placeholders